packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -6,372 +6,179 @@
 
 pragma solidity ^0.8.18;
 
-import {IProofVerifier} from "../ProofVerifier.sol";
 import {AddressResolver} from "../../common/AddressResolver.sol";
-import {LibAnchorSignature} from "../../libs/LibAnchorSignature.sol";
-import {LibBlockHeader, BlockHeader} from "../../libs/LibBlockHeader.sol";
-import {LibReceiptDecoder} from "../../libs/LibReceiptDecoder.sol";
-import {LibTxDecoder} from "../../libs/LibTxDecoder.sol";
-import {LibTxUtils} from "../../libs/LibTxUtils.sol";
-import {LibBytesUtils} from "../../thirdparty/LibBytesUtils.sol";
-import {LibRLPWriter} from "../../thirdparty/LibRLPWriter.sol";
+import {ChainData} from "../../common/IXchainSync.sol";
+import {LibTokenomics} from "./LibTokenomics.sol";
 import {LibUtils} from "./LibUtils.sol";
 import {TaikoData} from "../../L1/TaikoData.sol";
 
 library LibProving {
-    using LibBlockHeader for BlockHeader;
-    using LibUtils for TaikoData.BlockMetadata;
     using LibUtils for TaikoData.State;
 
-    bool private constant FLAG_VALIDATE_ANCHOR_TX_SIGNATURE = true;
-    bool private constant FLAG_CHECK_METADATA = true;
-    bool private constant FLAG_VALIDATE_HEADER_FOR_METADATA = true;
+    // keccak256("taiko")
+    bytes32 public constant VERIFIER_OK =
+        0x93ac8fdbfc0b0608f9195474a0dd6242f019f5abc3c4e26ad51fefb059cc0177;
```

Comment:
Had a comment here https://github.com/taikoxyz/taiko-mono/pull/13280#discussion_r1131320227, not sure if you saw.
> It should be possible to just do bytes32 public constant VERIFIER_OK =keccak256("taiko"); and the compiler will calculate it at compile time. Though I remember a couple of years ago it didn't always work reliably but should hopefully be working as expected in all cases.